src/actions/grant.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable import/prefer-default-export */
+import { GRANT_INPUT_CHANGED } from '../constants';
+import { log, logError } from '../logger';
+import { isDev } from '../config';
+
+export function inputChange(field, value) {
+  return {
+    type: GRANT_INPUT_CHANGED,
+    payload: {
+      [field]: value,
+    },
+  };
+}
+
+export function submitGrant() {
+  return async (dispatch, getStore, { graphqlRequest }) => {
+    const {
+      grant,
+      grant: {
+        name,
+        phone,
+        email,
+        date,
+        summary,
+        description,
+        askAmount,
+        totalCost,
+        googleToken,
+      },
+    } = getStore();
+
+    // @TODO do better validation
+    let validationError = null;
+    ['name', 'phone', 'email', 'date', 'summary', 'description', 'askAmount', 'totalCost', 'googleToken'].forEach(field => {
+      // Skip googleToken in dev mode
+      if (isDev && field === 'googleToken') return;
+      const value = grant[field];
+      if (!value || value === '') {
+        validationError = `${field} has wrong value`;
+      }
+    });
+
+    // @TODO handle error
+    if (validationError) return logError(validationError);
+
+    // @TODO handle error
+    if (date.unix() < (Date.now() / 1000)) return logError('Past date');
+
+    const query = `
+      mutation{
+        createSubmission(
+          name:"${name}",
+          phone:"${phone}",
+          email:"${email}",
+          date:"${date.format('DD/MM/YYYY')}",
+          summary:"${summary}",
+          description:"${description}",
+          askAmount:"${askAmount}",
+          totalCost:"${totalCost}",
+          googleToken:"${!isDev ? googleToken : ''}",
+        ) {
+          id
+        }
+      }
+    `;
+    log('query is', query);
+
+    try {
+      const { data, error } = await graphqlRequest(query);
```

Comment:
You are right, graphql exports `.errors` like this:
```
{
  "data": {
    "createSubmission": null
  },
  "errors": [
    {
      "message": "Validation error: Validation isEmail failed",
      "locations": [
        {
          "line": 1,
          "column": 10
        }
      ],
      "path": [
        "createSubmission"
      ]
    }
  ]
}
```
We also might want to change how we handle errors in the grapqhlRequest helper itself: https://github.com/CommunityFundIS/web/blob/cd325b88a347b7d588895518393c0729bdd0c028/src/store/createHelpers.js#L3-L18